protocols/luci-proto-batman-adv/htdocs/luci-static/resources/protocol/batadv_hardif.js

```diff
@@ -0,0 +1,60 @@
+'use strict';
+'require uci';
+'require form';
+'require network';
+
+network.registerPatternVirtual(/^bat.+$/);
+
+return network.registerProtocol('batadv_hardif', {
+	getI18n: function() {
+		return _('B.A.T.M.A.N. advanced Interface');
+	},
+	
+	getIfname: function() {
+		return this._ubus('l3_device') || this.sid;
+	},
+	
+	getOpkgPackage: function() {
+		return 'batman-adv';
+	},
+	
+	isFloating: function() {
+		return false;
+	},
+	
+	isVirtual: function() {
+		return false;
+	},
+	
+	getDevices: function() {
+		return null;
+	},
+	
+	containsDevice: function(ifname) {
+		return (network.getIfnameOf(ifname) == this.getIfname());
+	},
+	
+	renderFormOptions: function(s) {
+		var dev = this.getL3Device() || this.getDevice(), o;
+		
+		
+		// "master" is the batadv interface that we are linking to
+		
+		o = s.taboption('general', form.ListValue, 'master', _('B.A.T.M.A.N. interface'),_('This is the batman-adv device where you want to link the physical Device from above to. If this list is empty, then you need to create one first. If you want to route mesh traffic over a wired network device, then please select it from the above Device selector. If you want to assign the batman-adv interface to a Wi-fi mesh then do not select a Device in the Device selector but rather go to the Wireless settings and select this Interface as a network from there.'));
```

Comment:
Can you please break the line after the last comma? it's quite long.

Reply:
OK